jykuo-love-shiritori/twp
# [frontend] Page component

Resolves #146 

### Changes
Pages:
- [x] Manage User (limit: 10)
- [x] admin manage coupon  (limit: 6)
- [x] seller manage coupon  (limit: 6)
- [x] seller manage product (limit: 8)
- [x] buyer view shop coupon (limit: 12)
- [x] buyer view shop product (limit: 8)
- [ ] credit card (NO)
- [x] search product (limit: 8)
- [x] search shop product (limit: 6)
- [ ] search shop (obsolete)
- [x] order history  (limit: 4)
- [x] shipment (limit: 4)
- [x] discover (limit: 12)

Page Component function
- button ``<``, ``>`` that can naigate to prev/next page
  - will prevent going to invalid page ( less than 1 or no more result )
- ( i removed "jump to" function, it was very janky because we dont get the maximun results from backend )

### Testing Approach (if unconventional)
- go to the pages listed above and click the page compoent
- add more / delete items to see if the page compoent still function

```sh
docker compose up --build
```
### Additional Context
- pages have little difference, so i suggest test these i list below (they are more diff in implementation)
  - these api return obj that contain an array
    - buyer view shop product
  - these are search and u should test them very hard
    -  search product, search shop product
  - others are very similar. u can just select one to try to find bugs (other page will very likely to have same bugs if one exist)
